students/erik_oien/lesson05/mailroom3.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env python3
+
+import sys 
+import os
+
+donor_dict = {
+    "donor1": {"first_name": "Nathan", "last_name": "Explosion", "donations": [39595, 35081, 93295]}, 
+    "donor2": {"first_name": "Skwisgaar", "last_name": "Skwigelf", "donations": [37198]}, 
+    "donor3": {"first_name": "Toki", "last_name": "Wartooth", "donations": [20037, 32892, 99788]},
+    "donor4": {"first_name": "William", "last_name": "Murderface", "donations": [87470, 86870, 4397]},
+    "donor5": {"first_name": "Pickles", "last_name": "", "donations": [87838, 60282, 26653]},
+}
+
+# main menu for program 
+
+def menu():
+    while True:
+        welcome_input = input(opening_promt)
+        try:
+            main_menu_switch[welcome_input]()
+        except(KeyError):
+            print("Invalid option, please choose a number from the list below:")
+
+# thank you menu
+def thank_you():
+    while True:
+        donor_name = input(f"Please choose a donor or type list to see previous donors.\n{sub_menu_prompt}")
+        past_donor = False
+        if donor_name.lower() in ["menu", "quit"]:
+            sub_menu_switch[donor_name.lower()]()
+        elif donor_name.lower() == 'list':
+            past_donor = False
+            for donor in donor_dict.items():
+                print(donor[1]["first_name"] + " " + donor[1]["last_name"])
+        else:
+            try:
+                for donor in donor_dict.items():
```

Comment:
see my comment about your data structure - if you actually change it to utilize it then you can just do `donor_dict.get(donor_name)`